### ISSUE TITLE
Scanline Problems on Amiga

### ISSUE THREAD

**Issue by SirCathal** [c0pperdragon/Amiga-Digital-Video#16]
Hi there,
I use the adapter in my Amiga 500 and am very enthusiastic about it.
I only have one problem with the scanline option.
I activated it in the Preferences menu and then changed the "FB Size" from normal to double height in the Geometry menu.
Before, I had a sensational picture (ignoring the distortions from the cell phone camera)
But after the change, I have a totally shaky picture.
The photo is only a snapshot, in real life it is constantly shaking.
Am I doing something wrong or is it a general problem?
Maybee @IanSB knows about this?
![ScanlineSettings](https://user-images.githubusercontent.com/69156692/103490435-fd8ea680-4e1b-11eb-80ad-1b0df20476a9.jpg)
![FB Size Normal](https://user-images.githubusercontent.com/69156692/103490437-01bac400-4e1c-11eb-9519-583b8bc8d111.jpg)
![FB Size Double Height](https://user-images.githubusercontent.com/69156692/103490439-041d1e00-4e1c-11eb-8d1f-df678a0399ee.jpg)

**Comment by IanSB** [comment]
> I activated it in the Preferences menu and then changed the "FB Size" from normal to double height in the Geometry menu.

Currently scanlines require double height enabled and switching that on means that there are two uncached memory writes per pixel instead of one. Unfortunately in 12 bits per pixel mode which the Amiga requires there isn't enough system bandwidth for the double writes so you get the above messed up display as pixels are being dropped. 

You might be able to work around the problem by overclocking the Pi zero:
Go into the settings menu and try adjusting the overclock values, the core setting usually makes the most difference followed by the cpu setting, the sdram setting doesn't seem to make any difference. If those values are too high, some Pi zeros will freeze so only save the settings if the system appears to be stable. 
Note if you have saved overclock settings that cause the zero to hang up before you can clear them you will have to manually delete the saved profiles folder on the SD card to recover normal operation.

It should be possible to implement scanlines in an alternative way which doesn't require the double height option set and it's on my list of things to look at.

**Comment by SirCathal** [comment]
@IanSB Another solution for Scanlines would be great. On a big screen, scanlines are better for me :-)
But until then I will try the overclocking and hope that it will help.

**Comment by SirCathal** [comment]
Hi @IanSB 
Today I tested the overclocking and I could stabilise the Picture a bit, but not enough that I could live with it.   
The interesting part is, I set Overclock Core to 90. If I go higher, the Picture flickers more.
The other two Values don't have a significant impact. 
I even tried to set everything to the maximum value, but it dosen't  get better than the 0 - 90 - 0 setting in the Picture below.

![Overclocking](https://user-images.githubusercontent.com/69156692/103788059-b5030300-503e-11eb-8094-e6e02c578b03.jpg)

The other thing is, I had set the Scanline Level to 8 because I like scanlines that are halve bright compared to the normal lines.
But they showed completely black I tried lower and higher settings, but they are always only black.
Here are the result (In real live, the black lines are better visible than on the picture):
![ScanlinesZoom](https://user-images.githubusercontent.com/69156692/103789859-c4834b80-5040-11eb-99d4-79f1dc20f5a7.jpg)
![Scanlines](https://user-images.githubusercontent.com/69156692/103789878-ca792c80-5040-11eb-9f95-e8cf98919b5b.jpg)

**Comment by IanSB** [comment]
@SirCathal 

There is a download link for a beta test version in the hdmi capture card thread with scanlines support:
https://github.com/c0pperdragon/Amiga-Digital-Video/issues/19
Note that scanlines can only be black because adjustable scanlines require the two writes which as you see above has problems due to bandwidth issues.
It may be possibe to improve on this by using the GPU but there isn't much documentation for that so it may be a while before I can look at that.
Also note the link for a micro SD card extender to make software upgrades easier.

**Comment by IanSB** [comment]
Updated to beta10 due to cache alignment timing issue
https://github.com/IanSB/RGBtoHDMI/releases/download/626a527/beta10.zip

**Comment by SirCathal** [comment]
@IanSB Thank you, I will try this as soon as possible. My ordered Micro SD Extender should arrive today, then I will fit both together because it's a bit complicate to open that special Amiga Case. With the Micro SD Adapter I have to do that only once (thanks for the tip)

**Comment by SirCathal** [comment]
Hi @IanSB 
there is one thing I don't understand, you mentioned that the lines have to be written double with adjustable scanlines.
But isn't that done too when no scanlines are selected? 
Because with no scanlines, the original line is there twice too. Or is that al done inside the scaler?
If its done in the scaler, would it be possible to do the scanlines before the image is scaled? 
Or maybe only on none interlace screens? 
What I mean is, a 640x512 Pixel Screen without scanlines work fine but should need exact the same amount of pixels written then a 640x256 Screen with scanlines.
So maybe it could be a possible solution to allow the adjustable scanlines only with non interlace inputs?

**Comment by IanSB** [comment]
@SirCathal 

>Because with no scanlines, the original line is there twice too. Or is that all done inside the scaler

The video is captured at 640x256 and written to a 640x256 buffer in normal mode. The line doubling is done by the hardware scaler in the GPU but it can only replicate the line with the default scaler options.

In double height mode, the video is captured at 640x256 but written to a 640x512 buffer with two writes per pixel. The second pixel can have changed intensity for scanlines. For black scanlines, only 1 pixel is written and the other pixel is always left black.

It should be possible to get the GPU scaler to generate scanlines but that requires detailed documentation on writing display lists for the Pi in bare metal and there isn't much of that available at the moment.

I have managed to get 50% scanlines working in 12bpp mode. There is not enough processing time for fully variable and even then the timing is marginal and the Pi is very overclocked:
![capture112](https://user-images.githubusercontent.com/45300594/104348803-0eb17480-54fa-11eb-98b2-7baf8698e6a0.png)


However, it only seems to work properly with the CPLD version of RGBtoHDMI (screencap is from that) and with c0pperdragon's board I get glitches due to pixel drops. I think this is be because the pixel clock on the Amiga is slightly asymmetric which means less processing time on alternate pixels and c0pperdragon's board has to use that as it's clock. The CPLD version regenerates the pixel clock in the CPLD from the Hsync pulses and that is always perfectly symmetric so equal time on every pixel.

I'll post another beta soon with this change so you can try it on your system. You might get it to work with higher overclocking if you put a heasink on your Pi.

**Comment by SirCathal** [comment]
That seems that you are getting closer to a good solution, that's great and your image above looks perfect for me.
50% scanlines would be perfect, It's the setting I always use if it's available.

I will try to overclock as much as possible, I have some small heatsinks here to do that. 

I didn't know that there are two different clocks, I always taught the signal is synced to the sync signal from the Denise Chip in the Amiga.
But on the other hand, the Board has a Jumper to select between the different OCS and ECS Denise chip versions and I was wondering about that, because I always taught that those chips should produce completely identical signals as long as no ECS video mode is used. 

The only bad thing is that the delivery of my micro sd extender was shifted from today to tomorrow. So I hope I can test tomorrow after work.

**Comment by SirCathal** [comment]
Ah OK the resolutions.
I have monitors with 1280x1024 and 1920x1080 and a 4K TV. But no 1600x1200 or 1920x1200 monitor.
Maybe it work with the 1280x1024 monitor

**Comment by IanSB** [comment]
> 1440x1080 is a common resolution for older videos that could be handled by the most 1080p monitors

Yes, that works with no glitches:
![capture7](https://user-images.githubusercontent.com/45300594/104392640-6a035700-553a-11eb-9912-909da2d2398b.png)

If your monitor has a 1:1 pixel mode it should be able to display 1440x1080 with no problem, however not all monitors can do that so I have also added a special resolution which puts overscan padding of 240 pixels on either side so it looks like a 1920x1080 signal.

Unfortunately there is a bug in the Pi's nearest neighbour scaling kernel when using the overscan padding as the scaled output is a little mangled if you examine it closely. If I switch to interpolated scaling it looks OK.

**Comment by IanSB** [comment]
@SirCathal 

>So I hope I can test tomorrow after work.

Here is the new beta version:

https://github.com/IanSB/RGBtoHDMI/releases/download/c673ab0/beta11.zip

Wipe and reinstall on the SD card and reboot the Pi from the info menu or repower the Amiga

To make changing the settings easier I have added two extra profiles.
You can now change the Profile setting between:
Amiga
Amiga 0% scanlines
Amiga 50% scanlines

Amiga 0% scanlines is the same as the default Amiga profile but with just the scanlines setting turned on
This one should work at all resolutions

Amiga 50% scanlines has scanlines turned on, double height enabled and CPU and Core overclocked to 50 and 160 respectively.
They may need to go higher but that's a good starting point.
This profile will glitch in 1920x1080 mode but will give a stable image in 1280x1024, 1440x1080 and 1600x1200 modes

There are two new resolutions you can manually select:
1440x1080 - This is the preferred option but if your monitor can't display it at 1:1 pixels it may be unusable (You may need to change settings in your monitor to get it displaying 1:1)

1440(1920)x1080 - This is an experimental resolution which outputs 240 overscan padding pixels on either side of the 1440 image making a total of 1920 pixels. This should be treated by the monitor as 1920x1080 so no setting change should be required. However integer scaling doesn't work properly with this mode which I think is a bug in the Pi's boot firmware. If you compare the menus with a normal 1920x1080 image the text is a bit ragged/uneven even though all the ratios are correct for even scaling.

**Comment by SirCathal** [comment]
@IanSB WOW that works perfectly for me.
1440x1080 is pixel perfect on my monitor with the 50% setting.

The only thing that maybe could be a Problem is,  as long as I chose a fix resolution, there is no 50/60Hz auto switching.
For me personal, that's no problem because I always use 50Hz with the Amiga.

Thank you for your new release!

I will make more tests tomorrow and post some Pictures. Today it's a bit late here and I have to get up early for work.

**Comment by SirCathal** [comment]
I Made more tests and the results are fantastic.

Here some examples with 1280x1024 as output resolution and 50% Scanlines:
![capture1](https://user-images.githubusercontent.com/69156692/104655379-fa1dd980-56bd-11eb-9068-51a94862186e.png)
![capture2](https://user-images.githubusercontent.com/69156692/104655382-fab67000-56bd-11eb-8ece-d187d61413d7.png)
![capture3](https://user-images.githubusercontent.com/69156692/104655383-fb4f0680-56bd-11eb-9681-a125e86fe737.png)
![capture4](https://user-images.githubusercontent.com/69156692/104655384-fb4f0680-56bd-11eb-96f4-0750901f0844.png)
![capture5](https://user-images.githubusercontent.com/69156692/104655386-fb4f0680-56bd-11eb-986c-3fc1b7ec7ca6.png)
![capture6](https://user-images.githubusercontent.com/69156692/104655387-fb4f0680-56bd-11eb-90e3-4f9087fdccbf.png)
![capture7](https://user-images.githubusercontent.com/69156692/104655388-fbe79d00-56bd-11eb-89d0-bdf800b6b1bd.png)
![capture8](https://user-images.githubusercontent.com/69156692/104655389-fbe79d00-56bd-11eb-9529-4829b0c21753.png)
![capture9](https://user-images.githubusercontent.com/69156692/104655390-fbe79d00-56bd-11eb-9706-61ae42b996b6.png)
![capture10](https://user-images.githubusercontent.com/69156692/104655391-fc803380-56bd-11eb-95d4-aecdb6d0bef4.png)
![capture11](https://user-images.githubusercontent.com/69156692/104655393-fc803380-56bd-11eb-9b09-748d69c62bb8.png)
![capture12](https://user-images.githubusercontent.com/69156692/104655395-fc803380-56bd-11eb-9a55-7630d2f8252f.png)


And here 1440x1080 as output resolution and 50% Scanlines (Supporting Overscan):
![capture0](https://user-images.githubusercontent.com/69156692/104655448-0dc94000-56be-11eb-8388-64bb9a9721a1.png)
![capture14](https://user-images.githubusercontent.com/69156692/104655450-0dc94000-56be-11eb-8637-f0bc1ccf78fd.png)
![capture15](https://user-images.githubusercontent.com/69156692/104655452-0e61d680-56be-11eb-9930-b8288aba189f.png)
![capture16](https://user-images.githubusercontent.com/69156692/104655453-0e61d680-56be-11eb-80ba-62a75c7494b8.png)
![capture17](https://user-images.githubusercontent.com/69156692/104655454-0e61d680-56be-11eb-82f2-710468e5c828.png)
![capture18](https://user-images.githubusercontent.com/69156692/104655455-0efa6d00-56be-11eb-849e-263d7af78cd7.png)
![capture19](https://user-images.githubusercontent.com/69156692/104655456-0efa6d00-56be-11eb-900e-8cafd34c0a8c.png)
![capture20](https://user-images.githubusercontent.com/69156692/104655457-0efa6d00-56be-11eb-852e-14421f0f5ace.png)
![capture21](https://user-images.githubusercontent.com/69156692/104655460-0f930380-56be-11eb-8213-13f5a2f3adc3.png)
![capture22](https://user-images.githubusercontent.com/69156692/104655461-0f930380-56be-11eb-845e-4304dfb21bc8.png)
![capture23](https://user-images.githubusercontent.com/69156692/104655463-0f930380-56be-11eb-96d8-63571ebc9abc.png)
![capture24](https://user-images.githubusercontent.com/69156692/104655465-102b9a00-56be-11eb-9eaf-cd6ac2548dae.png)
![capture25](https://user-images.githubusercontent.com/69156692/104655466-102b9a00-56be-11eb-953a-c8af4df17709.png)
![capture26](https://user-images.githubusercontent.com/69156692/104655467-102b9a00-56be-11eb-9fc6-fca2ef47d11a.png)
![capture27](https://user-images.githubusercontent.com/69156692/104655468-102b9a00-56be-11eb-8c37-a075a682e6c6.png)
![capture28](https://user-images.githubusercontent.com/69156692/104655469-10c43080-56be-11eb-9965-08163a6cdb4e.png)
![capture30](https://user-images.githubusercontent.com/69156692/104655471-10c43080-56be-11eb-8216-1fb8a4285eb4.png)
![capture31](https://user-images.githubusercontent.com/69156692/104655472-10c43080-56be-11eb-8a41-e463abdfa3c6.png)
![capture32](https://user-images.githubusercontent.com/69156692/104655474-115cc700-56be-11eb-8d4b-8da954561bf1.png)
![capture33](https://user-images.githubusercontent.com/69156692/104655475-115cc700-56be-11eb-96f5-73abef3c2544.png)
![capture34](https://user-images.githubusercontent.com/69156692/104655476-11f55d80-56be-11eb-81b9-646fcd044c83.png)
![capture35](https://user-images.githubusercontent.com/69156692/104655477-11f55d80-56be-11eb-893b-f4a9596ba2c7.png)
![capture36](https://user-images.githubusercontent.com/69156692/104655478-11f55d80-56be-11eb-8c7b-20ee319ca973.png)
![capture37](https://user-images.githubusercontent.com/69156692/104655480-128df400-56be-11eb-9d68-bfa6172743d4.png)
![capture38](https://user-images.githubusercontent.com/69156692/104655481-128df400-56be-11eb-86fb-0a78ef85c360.png)
![capture39](https://user-images.githubusercontent.com/69156692/104655483-128df400-56be-11eb-9836-5a178342a92c.png)
![capture40](https://user-images.githubusercontent.com/69156692/104655484-13268a80-56be-11eb-981b-1fb5942a8d1e.png)
![capture41](https://user-images.githubusercontent.com/69156692/104655487-13268a80-56be-11eb-915f-604bd1e976f4.png)
![capture42](https://user-images.githubusercontent.com/69156692/104655489-13268a80-56be-11eb-9a75-41240d2e7433.png)
![capture43](https://user-images.githubusercontent.com/69156692/104655490-13268a80-56be-11eb-816d-befd352ff6b5.png)
![capture44](https://user-images.githubusercontent.com/69156692/104655491-13bf2100-56be-11eb-9285-61cdb8fa10fd.png)
![capture45](https://user-images.githubusercontent.com/69156692/104655493-13bf2100-56be-11eb-8ae0-b917f3dc3046.png)
![capture46](https://user-images.githubusercontent.com/69156692/104655496-13bf2100-56be-11eb-98e3-cac79080dde5.png)
![capture47](https://user-images.githubusercontent.com/69156692/104655498-1457b780-56be-11eb-8c6c-1b69dd77b1ad.png)
![capture48](https://user-images.githubusercontent.com/69156692/104655500-1457b780-56be-11eb-8e83-b8e613ad9089.png)
![capture49](https://user-images.githubusercontent.com/69156692/104655501-1457b780-56be-11eb-929d-78a2b1ce9d8e.png)
![capture50](https://user-images.githubusercontent.com/69156692/104655503-14f04e00-56be-11eb-8157-c53b7f122eaf.png)
![capture51](https://user-images.githubusercontent.com/69156692/104655505-14f04e00-56be-11eb-8dca-2aa9fa77b5e5.png)
![capture53](https://user-images.githubusercontent.com/69156692/104655506-14f04e00-56be-11eb-9d86-e4a08e5b1cba.png)
![capture54](https://user-images.githubusercontent.com/69156692/104655507-1588e480-56be-11eb-8f50-5425069e8764.png)

**Comment by IanSB** [comment]
> I Made more tests and the results are fantastic.

They look great!

Have you tried the "Swap aspect 625-525" setting  in the preferences menu. That will stretch NTSC sourced images so the above sources with the black border at the bottom should become full screen.

Also in the preferences menu is the screencap size option. By default screencaps are usually half size as that looks better when posting them but you can switch to full size to get 1440x1080 instead of 720x540. 

I fixed the 1440 with 1920 pixel padding mode so that now works as well which means 1440x1080 can now work on all 1080p TVs and monitors (e.g. my TV didn't support 1:1 pixel mode so would always stretch 1440 to full width).

I also added a new option for up to 5 saved configurations for each profile so you can quickly switch between different sets of preferences.

I'll post an updated beta soon.

**Comment by SirCathal** [comment]
@IanSB  I did not test the swap aspect option, but I will do more testing over the Weekend, today I had not the Time for testing but I will continue tomorrow.

**Comment by IanSB** [comment]
@SirCathal 

Here is another beta version (beta12):

https://github.com/IanSB/RGBtoHDMI/releases/download/48dff12/beta12.zip

This has resolution and refresh split into two options which makes it easier to change settings
Also new multiple saved configurations (5)
Automatic overscan to eliminate glitches in 50% scanline mode so you no longer have to select 1440x1080 and can leave it on Auto and it should still work.
Also has various bug fixes.

**Comment by SirCathal** [comment]
Hi @IanSB
Sorry for not being able to test a lot. Unfortunately, I got sick, but I assume that it will get better shortly and then I will test more.
But I tested something. The 625-525 aspect ratio menu item works fine. For me who is used to seeing a black stripe at the bottom of NTSC games, this is really unusual :-) but cool.
The scanlines also work in the auto setting at 1920x1080. however, from time to time I have a slight flicker or twitch in the picture. If I continue to overclock to 60/175/0 it is gone, but then the Pi crashes after a few minutes. But I don't have a heatsink on it yet. Therefore I have set 1440x1080 again at the moment.
What works great is the 50/60 Hz switching.
I've heard from other users that they would like another mode with 75% scanlines. would that still be possible?

**Comment by IanSB** [comment]
@SirCathal 

I hope you get better soon.

>The scanlines also work in the auto setting at 1920x1080. however, from time to time I have a slight flicker or twitch

When using 1920x1080 I overscan limited the picture to 1600x1080 as that stopped most of the problem and some profiles used that wider image but it looks like the extra little bit of overscan to 1440 is needed for the Amiga.

I have now changed it to overscan to 1440x1080 if the max width is set to 720 or less in the geometry menu (which it is with the default Amiga profile). This output will now be identical to the 1440(1920)x1080 resolution in the above beta 12 (but not the same as the "1440x1080" profile which has no overscan). 

When you feel better can you switch to "1440(1920)x1080" resolution with the above beta 12 and see if there are any problems?

**Comment by SirCathal** [comment]
Hi @IanSB
I tested 1440(1920)x1080 and it woks fine, no flickering great picture. It look's exactly like 1440x1080.
For me it's perfect.

I Made some Screenshots with and without the aspect scaling. 
Most Time it works fine, a few games stretches to far (for example Blood Money ) I think it uses some vertical ntcs overscan but I'm not sure.
The Blood Money Intro stretches fine:
![capture11](https://user-images.githubusercontent.com/69156692/105420717-29de5b80-5c41-11eb-816d-f41a965f5118.png)
![capture12](https://user-images.githubusercontent.com/69156692/105420719-2a76f200-5c41-11eb-91c9-38600be416c4.png)

The Game itself expands the vertical resolution a bit:
![capture14](https://user-images.githubusercontent.com/69156692/105420787-437fa300-5c41-11eb-8026-b0452bd822e0.png)

The scaling of Monkey Island and Sysinfo is fine:
![capture9](https://user-images.githubusercontent.com/69156692/105420897-788bf580-5c41-11eb-9058-37a24f84ca41.png)
![capture10](https://user-images.githubusercontent.com/69156692/105420898-79248c00-5c41-11eb-97a6-50b05c2609be.png)
![capture23](https://user-images.githubusercontent.com/69156692/105420929-85a8e480-5c41-11eb-9a44-9fd8b563cb98.png)
![capture22](https://user-images.githubusercontent.com/69156692/105420937-88a3d500-5c41-11eb-95b4-bf422a1c3c8c.png)

**Comment by SirCathal** [comment]
@IanSB If I can something else for you please let me know.
And do you think you can add one more profile with 75% scanlines?

**Comment by IanSB** [comment]
@SirCathal 

>If I can test something else for you please let me know.

Here is beta14 with various fixes including Auto resolution reducing the width to 1440 so can you check that works for you:

https://github.com/IanSB/RGBtoHDMI/releases/download/98d62b35/beta14.zip

>And do you think you can add one more profile with 75% scanlines?

I may be able to add 25% and 75% scanlines but it will require some rewriting and code optimisation so it may be a while before that happens.

**Comment by SirCathal** [comment]
@IanSB I tested beta 14 and it works very good.
Auto resolution detects my 1920x1080 display and uses 1440 (displaying 1920x1080) but there is no flickering in 50% scanline mode so it must be 1440x1080.
![capture1](https://user-images.githubusercontent.com/69156692/105525499-78433700-5ce1-11eb-8fe7-5d4034721495.png)

**Comment by IanSB** [comment]
@SirCathal 

https://github.com/IanSB/RGBtoHDMI/releases

Beta 16 now has fully variable scanlines after I changed the frame buffer format.
Can you test to see if it is OK for you.

I decreased the overclock settings because the code is now more efficient but if you see any issues, try increasing again.

**Comment by SirCathal** [comment]
@IanSB 
Thanks for the new beta. I tested the new variable scanlines and in general they work great.
They work fine with auto resolution 

But I also found some issues.
1)	When I start a NTSC WHDLoad Game on my Pal Amiga (for example Frontier Elite 2) it switches correctly to 60Hz. But forgot my Scanline Level (I selected 12 and saved that configuration) but when it switches to 60Hz it steps back to the default of 6
2)	After closing the Game, the workbench output is still 60Hz and the lower part of the screen is cut of because it’s a Pal screen

**Comment by IanSB** [comment]
>When I start a NTSC WHDLoad Game on my Pal Amiga (for example Frontier Elite 2) it switches correctly to 60Hz. But forgot my Scanline Level (I selected 12 and saved that configuration) but when it switches to 60Hz it steps back to the default of 6

That is expected because there are separate 50Hz and 60Hz sub-profiles and it will auto select the 60Hz sub-profile when the timing changes. When a profile or sub-profile changes it loads it's own setting so you have to change the scanline level on that sub-profile and save the config. After that it should switch between the two profiles with the right scanline level.

To confirm this, look at the Sub-profile line on the main menu, it should have changed from Amiga 50Hz to Amiga 60Hz PAL

There are actuall 3 sub-profiles:
Amiga 50Hz          - normal 50Hz
Amiga 60Hz          - normal 60Hz
Amiga 60Hz PAL  - 60Hz but from a PAL amiga so it used the PAL crystal which means different timings from real NTSC

>After closing the Game, the workbench output is still 60Hz and the lower part of the screen is cut of because it’s a Pal screen

If the computer is outputting 50Hz timings it should reload the 50Hz sub-profile and revert to full screen.
When this happens, can you post a photo of the source summary option in the info menu and also let me know which sub-profile (not profile) is showing on the main menu.

**Comment by SirCathal** [comment]
@IanSB 
Yes you are right, I can save the settings for 50 and 60 Hz and then it works.
The other thing was a bad WHDLoad Slave. It sets NTSC but don't changed it back to PAL.
When I use the NTSC tooltype with the WHDLoad title it works correctly. Starting in Pal (Workbench) start the Game (switch to NTSC) and after exiting the game it switches back to PAL.

For me that version is great. All scanline levels work fine and without any flickering with auto resolution up to 1920x1080. 

I found only one small issue with the auto resolution, but that's very special to my monitor setup.
I use a active HDMI switch and with one of that ports I connected another active HDMI switch.

If I don't select the correct input port on both switches, Auto resolution fails and I get no Picture.
I think that's because in that moment, there is no monitor to detect.
Maybe a default resolution could help in that case. But overall it's not that important.
As long as I do not test, I simply select 1440x1080 and have no Problem.
